### PR TITLE
JAMES-3445 James Webadmin Cli Domain Exist Command

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

File: server/protocols/webadmin-cli/src/main/java/org/apache/james/cli/domain/DomainExistCommand.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    @Override
+    public Integer call() {
+        try {
+            DomainClient domainClient = Feign.builder()
+                .target(DomainClient.class, domainCommand.webAdminCli.jamesUrl + "/domains");
+            Response rs = domainClient.isExist(domainName);
+            if (rs.status() == EXISTED_CODE) {
+                domainCommand.out.println(domainName + " exists");
+            } else {
+                domainCommand.out.println(domainName + " does not exist");
```
Comment (chibenwa) [inline server/protocols/webadmin-cli/src/main/java/org/apache/james/cli/domain/DomainExistCommand.java:52]: So if James sends back a 500 error code, you say to the user "the domain does not exist"?

File: server/protocols/webadmin-cli/src/main/java/org/apache/james/httpclient/DomainClient.java [hunk 2]
```diff
@@ -36,4 +36,7 @@
     @RequestLine("DELETE /{domainToBeDeleted}")
     Response deleteADomain(@Param("domainToBeDeleted") String domainName);
 
+    @RequestLine("GET /{domainName}")
+    Response isExist(@Param("domainName") String domainName);
```
Comment (chibenwa) [inline server/protocols/webadmin-cli/src/main/java/org/apache/james/httpclient/DomainClient.java:40]: doesExist

File: server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java [hunk 3]
```diff
@@ -121,4 +121,26 @@ void domainDeleteCommandWithDefaultDomainShouldFailed(GuiceJamesServer server) {
         assertThat(exitCode).isEqualTo(1);
     }
 
+    @Test
+    void domainExistCommandWithDefaultDomainShouldBeExisted(GuiceJamesServer server) {
```
Comment (chibenwa) [inline server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java:125]: Please commit test and code together ;-)

File: server/protocols/webadmin-cli/src/main/java/org/apache/james/httpclient/DomainClient.java (hunk 2 above)
Comment (quantranhong1999) [reply server/protocols/webadmin-cli/src/main/java/org/apache/james/httpclient/DomainClient.java:40]: ok, I think i should read more grammar about does Exist/ be Exist :3

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
`docker: Error response from daemon: Invalid container name (git-publish-feature/domainexist-961563a0-3c29-49a0-8838-993fc42603f1), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed.`

I will rebase this work, and propose it in a new branch, without the invalid characters...

**Comment by quantranhong1999** [comment]
> `docker: Error response from daemon: Invalid container name (git-publish-feature/domainexist-961563a0-3c29-49a0-8838-993fc42603f1), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed.`
> 
> I will rebase this work, and propose it in a new branch, without the invalid characters...

Because of "/" i think.
Should I create another PR with different branch instead?

**Comment by Arsnael** [comment]
```
[78f195547192fac4b8f2d1667bcc9e0b70ddbfc9] + docker run -a stdout -a stderr -t --name git-publish-feature/domainexist-961563a0-3c29-49a0-8838-993fc42603f1 -v /origin git/publish-feature/domainexist-961563a0-3c29-49a0-8838-993fc42603f1 78f195547192fac4b8f2d1667bcc9e0b70ddbfc9 MERGE-feature/domainexist-961563a0-3c29-49a0-8838-993fc42603f1 master
[78f195547192fac4b8f2d1667bcc9e0b70ddbfc9] docker: Error response from daemon: Invalid container name (git-publish-feature/domainexist-961563a0-3c29-49a0-8838-993fc42603f1), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed.
```
don't put a `/` in your branch names, the CI doesn't seem to like it unfortunately^^'

**Comment by chibenwa** [comment]
See https://github.com/linagora/james-project/pull/4003

**Pull request closed without merging**